### ISSUE TITLE
Alpha 8 - LED's retain last state after clearing a source

### ISSUE THREAD

**Issue by flexage** [hyperion-project/hyperion.ng#1007]
- [x] I confirm that this is an issue rather than a question.

#### Steps to reproduce

I'm using Apa102's driven from the pi 3, I also have a second instance on the same install using WLED but this doesn't seem to be affected).

**Test 1**

1. Start Hyperion (Rainbow Startup Animation Plays)
2. Rainbow startup animation finishes, and LED's turn off for a moment, but then come back with a frozen rainbow

**Test 2**

1. Set a colour using the Web UI Remote control (LED's successfully show the colour)
2. Clear the colour source set in the Web UI (the source disappears from the source panel, but the LED's remain the colour set)

**Test 3**

1. Set a colour using the Web UI Remote control (Led's show the colour ok)
2. Send a Protocol Buffers/Hyperion Android Grabber stream (Led's are matching the video source fine)
3. Stop the android grabber (Sources panel reflects Grabber source is gone but Led's freeze on last frame of stream, should be showing the lower priority colour source from the web ui now).

#### What is expected?

I expect that clearing the sources for effects, colours, and streams will also clear the LED's and not just the sources panel.

#### What is actually happening?

It looks like, with the APA102's at least, then when a source is cleared, it's final values are retained on the LED's - or sometimes cleared from the LED's momentarily, but then return.

#### System
Hyperion Server: 
- Build:       (HEAD detached at 08909b8a) (Paulchen-Panther-75963ca9/08909b8a-1600115237)
- Build time:  Sep 14 2020 21:04:51
- Git Remote:  https://github.com/hyperion-project/hyperion.ng
- Version:     2.0.0-alpha.8
- UI Lang:     auto (BrowserLang: en-GB)
- UI Access:   default
- Avail Capt:  dispmanx,v4l2,framebuffer,qt

Hyperion Server OS: 
- Distribution: Raspbian GNU/Linux 10 (buster)
- Arch:         arm
- Kernel:       linux (4.19.97-v7+ (WS: 32))
- Browser:      Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36

**Comment by Lord-Grey** [comment]
Hi @flexage 
thanks for your feedback. To track down the issue, may I ask you providing an export of your configuration:

via UI: Configuration->General->Import/Export Configuration-> click "Export"

Plus, could you set hyperion.ng in Debug mode logging and share the log output, please?

via UI: System->Log-> Set Log-Level = Debug

If you familiar doing things on the commandline, you might start hyperion in debug mode already and have the loglines at hand easily:

pkill hyperiond
hyperiond -d

Restart hyperion.ng, run Test 1 and 2 and share the log-output.
I assume all three test scenarios share the same root cause. Therefore, lets focus on the first two.

Thank you!

**Comment by Lord-Grey** [comment]
Hi @flexage

Just one additional question:

I assume that you have defined a "Refresh time for the LED device

![image](https://user-images.githubusercontent.com/48840279/93196267-7c1ae780-f74a-11ea-8d76-c4d8f7b7e40a.png)

Could you set it to zero and see, if the problem still exists.

![image](https://user-images.githubusercontent.com/48840279/93196442-bbe1cf00-f74a-11ea-9400-c179174640a7.png)

Just to ring-fence the problem.

Thank you!

**Comment by Lord-Grey** [comment]
Hi @flexage I think I found the error. Let me apply a fix and get back to you for testing.

**Comment by flexage** [comment]
Hi @Lord-Grey, great work! Sorry for not being quicker at getting back to you, I'll be available this evening (GMT) to provide any info and do any testing, please feel free to log anything you require here after looking at your fix and I'll be happy to provide it

**Comment by Lord-Grey** [comment]
@flexage May I ask you testing PR #1008 and report back, please?

**Comment by flexage** [comment]
@Lord-Grey Great work dude! Just tested the Debian RPi 2/3 build from the PR and that code change seems to have fixed it 👍 

I'll keep using this build for now, and let you know if I spot any related edge cases

**Comment by Lord-Grey** [comment]
@flexage Good to hear that it is working at your end. Thanks for your support bringing it up and doing the testing.

**Comment by flexage** [comment]
No problem, happy to help, thanks for all the great work on Hyperion

**Comment by flexage** [comment]
Sorry, it looks like I  confirmed this as fixed to soon...

I ended up staying on Alpha 6 for a while as it was the most stable for me, but tonight I have been trying Alpha 9.

I am still getting the last frame of the video effect/color/protobuffer output 'sticking' on my LED's when the source is cleared.

I checked the "Refresh Time" on the LED Device, as you mentioned above, and it was set to 5000 - must have been a default, potentially from a very old (pre-alpha) version, I had to enable expert settings to find it. I tried setting it to 0 but it didn't have an effect on the problem.
